x/ccv/provider/keeper/relay.go

```diff
@@ -387,11 +393,15 @@ func (k Keeper) HandleSlashPacket(ctx sdk.Context, chainID string, data ccv.Slas
 	)
 
 	// Obtain validator from staking keeper
-	validator, found := k.stakingKeeper.GetValidatorByConsAddr(ctx, providerConsAddr.ToSdkConsAddr())
+	validator, err := k.stakingKeeper.GetValidatorByConsAddr(ctx, providerConsAddr.ToSdkConsAddr())
+	if err != nil && errors.Is(err, stakingtypes.ErrNoValidatorFound) {
+		k.Logger(ctx).Error("validator not found or is unbonded", "validator", providerConsAddr.String())
+		return
+	}
 
 	// make sure the validator is not yet unbonded;
 	// stakingKeeper.Slash() panics otherwise
-	if !found || validator.IsUnbonded() {
+	if validator.IsUnbonded() {
```

Comment:
adjust the error message (I cannot comment directly on the line, its a few lines below here)